pkg/prometheus/promcfg.go

```diff
@@ -726,6 +730,7 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 	cfg = enforcer.addLimitsToYAML(cfg, labelLimitKey, m.Spec.LabelLimit, enforcedLabelLimit)
 	cfg = enforcer.addLimitsToYAML(cfg, labelNameLengthLimitKey, m.Spec.LabelNameLengthLimit, enforcedLabelNameLengthLimit)
 	cfg = enforcer.addLimitsToYAML(cfg, labelValueLengthLimitKey, m.Spec.LabelValueLengthLimit, enforcedLabelValueLengthLimit)
+	cfg = enforcer.addLimitsToYAML(cfg, bodySizeLimitKey, 0, enforcedBodySizeLimit)
```

Comment:
For consistency purposes, I think it would make sense, but it is still very unlikely that users will define this limit per ServiceMonitor since it is a very advanced configuration option.

Reply:
ya I felt that we won't be using it much even if we provide option in monitors and probe. But I am open to add it in them if we need to

Reply:
IMHO it's fine to wait and see if there is demand for adding it to service/pod monitors and probes since it's a new feature. And the main use case is to avoid Prometheus from blowing up when a target starts exposing hundreds megabytes of data by accident.